### ISSUE TITLE
I have filled up my redis db

### ISSUE THREAD

**Issue by kpturner** [capriza/node-busmq#21]
Not sure if I have had a bug or not, but I noticed that I have absolutely thousands of keys in my redis database associated with busmq.

The keys are all called "bus:queue:<unique_queue_name>:msgid" and have a value of 1.

What are they and, more importantly, how do I get rid of them?

Redis reports that they have a TTL of -1. When I attach to a queue I leave the defaults (30 seconds) but during testing I may well have had bugs where I attached but failed to detach.

**Comment by kpturner** [comment]
The sequence of events is that I have a client and a server (both child-processes).  

The client creates a uniquely named queue, attaches to it and consumes it.
It sends a message to the server - the message contains the name of the queue.
The server responds on the queue specified (with a push()) then instantly and detaches from it.
The client receives the message and also detaches from it:
rq.on("message",function(response,id){
    blah;
    rq.detach();
})

Each time, a single key is left in the redis DB containing the:
bus:queue:myuniqueq:msgid

**Comment by fujifish** [comment]
Is that the only busmq key that leaks?
Can you provide actual code that reproduces this behaviour?
What version of redis are you using?

**Comment by kpturner** [comment]
I will try to create a little test script that replicates it.

This is a windows 10 PC, so it uses Redis 2.8.2400

**Comment by kpturner** [comment]
Here is an app that will leak one key as described above

[busmq.zip](https://github.com/capriza/node-busmq/files/115878/busmq.zip)

**Comment by fujifish** [comment]
You are not detaching from outq in app.js. Is that the queue that's leaking?

**Comment by kpturner** [comment]
No it is not - it is the queue in am specifically detaching from that is leaking ("inq").  

The other queue disappears completely from redis 30 seconds after app.js has terminated.

**Comment by fujifish** [comment]
you indeed found a bug!
will be pushing a fix shortly. thanks!

**Comment by fujifish** [comment]
published version 1.1.3 with the fix. Please give it a try

**Comment by kpturner** [comment]
Perfect :)

Thanks
